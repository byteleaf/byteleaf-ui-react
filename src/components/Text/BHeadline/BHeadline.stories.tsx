import React from 'react';

import { BHeadline } from './BHeadline';
import { GrayBackground } from '../../../../.storybook/decorators';

export default {
  title: 'Text',
};

export const BHeadlineStory = () => (
  <BHeadline>This is a Headline with a small padding at the bottom (grey background for demonstration)</BHeadline>
);

BHeadlineStory.story = {
  name: 'BHeadline',
  decorators: [GrayBackground],
  parameters: {
    notes: 'A Headline component, for example used as a title on a Paper or Card.',
  },
};
