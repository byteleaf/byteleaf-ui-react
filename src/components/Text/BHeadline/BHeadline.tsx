import React from 'react';
import { styled, Typography } from '@material-ui/core';

const HeadlineWrapper = styled('div')(({ theme }) => ({
  paddingBottom: theme.spacing(2),
}));

export const BHeadline = ({ children }: { children: string }) => (
  <HeadlineWrapper>
    <Typography variant="h5" component="h3">
      {children}
    </Typography>
  </HeadlineWrapper>
);
