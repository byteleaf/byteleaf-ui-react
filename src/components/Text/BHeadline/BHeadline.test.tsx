import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BHeadline } from './BHeadline';

test('loads and displays headline story', async () => {
  const { container } = render(<BHeadline>Headline</BHeadline>);

  expect(container).toMatchSnapshot();
});
