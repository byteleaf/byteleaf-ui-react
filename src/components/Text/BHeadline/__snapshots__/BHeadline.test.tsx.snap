// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays headline story 1`] = `
<div>
  <div
    class="div-root-1 div-root-2"
  >
    <h3
      class="MuiTypography-root MuiTypography-h5"
    >
      Headline
    </h3>
  </div>
</div>
`;
